schemas/twitter.js

```diff
@@ -0,0 +1,167 @@
+import * as twitter from '../apis/twitter';
+
+import {
+    GraphQLSchema,
+    GraphQLObjectType,
+    GraphQLID,
+    GraphQLString,
+    GraphQLNonNull,
+    GraphQLInt,
+    GraphQLList,
+} from 'graphql';
+
+let UserType = new GraphQLObjectType({
+
+    name: 'TwitterUser',
+    description: 'Twitter user, or as we like to say in France: a \'tweetos\'',
+    fields: () => ({
+        created_at: { type: GraphQLString },
+        description: { type: GraphQLString },
+        id: { type: GraphQLID }, // GraphQLInt would return null
+        screen_name: { type: GraphQLString },
+        name: { type: GraphQLString },
+        profile_image_url: { type: GraphQLString },
+        url: { type: GraphQLString },
+        tweets_count: { 
+            type: GraphQLInt,
+            resolve: ({ statuses_count }) => statuses_count
+        },
+        followers_count: { type: GraphQLInt },
+        tweets: {
+            type: new GraphQLList(TweetType),
+            description: 'Get a list of tweets for current user',
+            args: {
+                limit: { type: GraphQLInt }
+            },
+
+            //             user            args
+            resolve: ({ id: user_id }, { limit = 10 }) => twitter.getTweets(user_id, limit)
+        }
+    })
+
+});
+
+let TweetType = new GraphQLObjectType({
+
+    name: 'Tweet',
+    description: 'A tweet object',
+    fields: () => ({
+        id: { type: GraphQLID },
+        created_at: { type: GraphQLString },
+        text: { type: GraphQLString },
+        retweet_count: { type: GraphQLInt },
+        user: {
+            type: UserType
+        },
+        retweets: {
+            type: new GraphQLList(RetweetType),
+            description: 'Get a list of retweets',
+            args: {
+                limit: { type: GraphQLInt }
+            },
+            //        passing integer 'id' here doesn't work surprisingly, had to use 'id_str'
+            resolve: ({ id_str: tweetId }, { limit = 5 }) => twitter.getRetweets(tweetId, limit)
+        }
+    })
+});
+
+let RetweetType = new GraphQLObjectType({
+
+    name: 'Retweet',
+    description: 'Retweet of a tweet',
+    fields: () => ({
+        id: { type: GraphQLID },
+        created_at: { type: GraphQLString },
+        in_reply_to_tweet_id: {
+            type: GraphQLString,
+            resolve: ({ in_reply_to_status_id }) => in_reply_to_status_id
+        },
+        in_reply_to_user_id: { type: GraphQLInt },
+        in_reply_to_screen_name: { type: GraphQLString },
+        retweeted_status: {
+            type: TweetType
+        },
+        user: {
+            type: UserType
+        }
+    })
+});
+
+let twitterType = new GraphQLObjectType({
+
+    name: 'TwitterAPI',
+    description: 'The Twitter API',
+    fields: {
+        user: {
```

Comment:
What you think about using a single arg wich is a custom GraphQLScalarType ?
https://facebook.github.io/graphql/#sec-Scalars
http://stackoverflow.com/questions/33819658/dynamic-unique-objects-in-graphql